core/bbox/bbox.go

```diff
@@ -0,0 +1,53 @@
+// Package bbox provides utilities for managing various types of bounding boxes.
+package bbox
+
+import (
+	"fmt"
+
+	"github.com/otrego/clamshell/core/point"
+)
+
+// New creates a new bounding box.
+func New(tl, br *point.Point) (*BoundingBox, error) {
+	if tl.X() >= br.X() || tl.Y() >= br.Y() {
+		// Open question: Should we allow 0-width / 0-height bboxes? It doesn't seem
```

Comment:
I don't think we should. If it shouldn't happen, we shouldn't let it happen.

Reply:
Removed comment.